toonz/sources/toonz/commandbarpopup.cpp

```diff
@@ -0,0 +1,480 @@
+#include "commandbarpopup.h"
+
+// Tnz includes
+#include "tapp.h"
+#include "menubar.h"
+#include "shortcutpopup.h"
+
+// TnzQt includes
+#include "toonzqt/gutil.h"
+
+// TnzLib includes
+#include "toonz/toonzfolders.h"
+
+// TnzCore includes
+#include "tsystem.h"
+
+// Qt includes
+#include <QMainWindow>
+#include <QPushButton>
+#include <QVBoxLayout>
+#include <QHBoxLayout>
+#include <QGridLayout>
+#include <QHeaderView>
+#include <QtDebug>
+#include <QXmlStreamReader>
+#include <QXmlStreamWriter>
+#include <QDataStream>
+#include <QMimeData>
+#include <QDrag>
+#include <QMouseEvent>
+#include <QPainter>
+#include <QApplication>
+#include <QLabel>
+
+//=============================================================================
+// CommandBarCommandItem
+//-----------------------------------------------------------------------------
+
+class CommandBarCommandItem final : public QTreeWidgetItem {
+  QAction* m_action;
+
+public:
+  CommandBarCommandItem(QTreeWidgetItem* parent, QAction* action)
+      : QTreeWidgetItem(parent, UserType), m_action(action) {
+    setFlags(Qt::ItemIsSelectable | Qt::ItemIsDragEnabled | Qt::ItemIsEnabled |
+             Qt::ItemNeverHasChildren);
+    setText(0, m_action->text().remove("&"));
+    setToolTip(0, QObject::tr("[Drag] to move position"));
+  }
+  QAction* getAction() const { return m_action; }
+};
+
+//=============================================================================
+// CommandBarSeparatorItem
+//-----------------------------------------------------------------------------
+
+class CommandBarSeparatorItem final : public QTreeWidgetItem {
+public:
+  CommandBarSeparatorItem(QTreeWidgetItem* parent)
+      : QTreeWidgetItem(parent, UserType) {
+    setFlags(Qt::ItemIsSelectable | Qt::ItemIsDragEnabled | Qt::ItemIsEnabled |
+             Qt::ItemNeverHasChildren);
+    setText(0, QObject::tr("----Separator----"));
+    setToolTip(0, QObject::tr("[Drag] to move position"));
+  }
+};
+
+//=============================================================================
+// CommandBarTree
+//-----------------------------------------------------------------------------
+
+CommandBarTree::CommandBarTree(TFilePath& path, QWidget* parent)
+    : QTreeWidget(parent) {
+  setObjectName("SolidLineFrame");
+  setAlternatingRowColors(true);
+  setDragEnabled(true);
+  setDropIndicatorShown(true);
+  setDefaultDropAction(Qt::MoveAction);
+  setDragDropMode(QAbstractItemView::DragDrop);
+  setIconSize(QSize(21, 17));
+
+  setColumnCount(1);
+  header()->close();
+
+  /*- Load path if it does exist. If not, then load from the template. -*/
+  TFilePath fp;
+  if (TFileStatus(path).isWritable())
+    fp = path;
+  else {
+    if (path.getName() == "xsheettoolbar") {
+      fp = ToonzFolder::getTemplateModuleDir() + TFilePath("xsheettoolbar.xml");
+    } else {
+      fp = ToonzFolder::getTemplateModuleDir() + TFilePath("commandbar.xml");
+    }
+  }
```

Comment:
- By default, `ToonzFolder::getTemplateModuleDir()` will return `stuff\profiles\layouts\settings` .
But for now `xsheettoolbar.xml` and `commandbar.xml` are under `stuff\profiles\layouts` so it fails to load the template properly.

- You can use `ToonzFolder::getModuleFile(TFilePath filename)` to obtain path to the setting files. This function will return a path to the personal setting (under `stuff\profiles\layouts\settings.username` ) if it exists, or return a path to the template setting if the personal one does not exist.

Reply:
Good catch. I fixed this and also moved the editinplace command to a toggle like you suggested.